README.md
# OXID cookie compliance modul for OXID 6

Free OXID eShop module for GDRP compliance.

Module for OXID 4/5: https://github.com/aggrosoft/oxid-cookie-compliance/tree/oxid-410

## Installation
- Get module by composer `composer require aggrosoft/oxid-cookie-compliance`

## Configuration
- See module settings
- Activate "Lern-Modus" and make an test order, after this deacticate "Lern-Modus"
- Change cookie settings in general shop data --> cookies
- Add `[{include file="widget/cookieinfos.tpl"}]` to `oxsecurityinfo` cms-page
- "Lern-Modus" will handle the oxid standard cookies (ESSENTIAL, Google Analytics)
- Add extra cookies in the backend module    
- Customize the including of the extra cookies in templates by using `[{if $oViewConf->isCookieCategoryEnabled('MARKETING') == 1}][{/if}]`
- Possible cats are ESSENTIAL,PERSONALIZATION,ANALYTICS,MARKETING,UNCATEGORIZED

## Notice
- Template compatibility only for wave theme, feel free to contribute for other themes ;-)
- If using more languages add table `compliancecookies` to `$this->aMultiLangTables` config.inc.php (supports oxtitle, oxdescription)


CHANGELOG.md
### Changelog

All notable changes to this project will be documented in this file. Dates are displayed in UTC.

Generated by [`auto-changelog`](https://github.com/CookPete/auto-changelog).

#### [1.2.4](https://github.com/aggrosoft/oxid-cookie-compliance/compare/1.2.3...1.2.4)

> 29 June 2022

- fix module setting banner position bottom right [`#29`](https://github.com/aggrosoft/oxid-cookie-compliance/pull/29)

#### [1.2.3](https://github.com/aggrosoft/oxid-cookie-compliance/compare/1.2.2...1.2.3)

- Bump node-fetch from 2.6.1 to 2.6.7 [`#26`](https://github.com/aggrosoft/oxid-cookie-compliance/pull/26)
- Bump hosted-git-info from 2.8.8 to 2.8.9 [`#21`](https://github.com/aggrosoft/oxid-cookie-compliance/pull/21)
- Bump path-parse from 1.0.6 to 1.0.7 [`#23`](https://github.com/aggrosoft/oxid-cookie-compliance/pull/23)
- increased z-index to 9999 for all [`#24`](https://github.com/aggrosoft/oxid-cookie-compliance/pull/24)
- implement (right) handling for opt-out consent type [`#22`](https://github.com/aggrosoft/oxid-cookie-compliance/pull/22)
- fix function definition [`0c4afb7`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/0c4afb7fecc245d774804afd9a300f1d13c2df55)
- add missing use statement [`c8e0cbd`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/c8e0cbd2e4ed665ec529debdfb4d2e7f85ce2026)

#### [1.2.2](https://github.com/aggrosoft/oxid-cookie-compliance/compare/1.2.1...1.2.2)

> 1 April 2021

- Release 1.2.2 [`a0fe2cf`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/a0fe2cfbebfc50ffec8cd1041e2829a231c8b51e)
- Missing quotes [`c12cbf0`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/c12cbf05608ffe9ba7ac42ff058ec3effd6a4c10)

#### [1.2.1](https://github.com/aggrosoft/oxid-cookie-compliance/compare/1.2.0...1.2.1)

> 25 March 2021

- added forgotten vanilla min file [`#20`](https://github.com/aggrosoft/oxid-cookie-compliance/pull/20)
- Release 1.2.1 [`a0ee469`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/a0ee469233578cca1823f4d6ab322ac1ea5ee634)
- added lang entry [`827959b`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/827959baa0547ec4f35325c4b3db249276699e40)

#### [1.2.0](https://github.com/aggrosoft/oxid-cookie-compliance/compare/1.1.11...1.2.0)

> 24 March 2021

- added vanilla javascript, added jquery switch, bunped dependecies, fi… [`#19`](https://github.com/aggrosoft/oxid-cookie-compliance/pull/19)
- added vanilla javascript, added jquery switch, bunped dependecies, fixed COOKIE_COMPLIANCE_URL not set [`c7663a5`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/c7663a550ac9cfb9feafdd42f27a33054d86314c)
- Release 1.2.0 [`4be1b4c`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/4be1b4cec553a5b5f8dd9f2b7d87ef8cf63e5214)
- fresh build [`0b2cf21`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/0b2cf21d8beab736bfaa0bc5c1021f9c59ea6ffb)

#### [1.1.11](https://github.com/aggrosoft/oxid-cookie-compliance/compare/1.1.10...1.1.11)

> 12 March 2021

- fix typo [`dc0ce78`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/dc0ce780f2b63ca0d764f69eee4d44e6efa3c1c0)

#### [1.1.10](https://github.com/aggrosoft/oxid-cookie-compliance/compare/1.1.9...1.1.10)

> 12 March 2021

- remove version [`eb88531`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/eb88531bf2aa4e97b28d31ce353e8865b38b0c6d)

#### [1.1.9](https://github.com/aggrosoft/oxid-cookie-compliance/compare/1.1.8...1.1.9)

> 12 March 2021

- Update agcookiecompliance_lang.php [`da7c956`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/da7c95620900c9c05de58590d84acde0afd9009b)

#### [1.1.8](https://github.com/aggrosoft/oxid-cookie-compliance/compare/1.1.7...1.1.8)

> 11 March 2021

- add fr lang [`#17`](https://github.com/aggrosoft/oxid-cookie-compliance/pull/17)
- add en lang [`#16`](https://github.com/aggrosoft/oxid-cookie-compliance/pull/16)
- multilang info [`#15`](https://github.com/aggrosoft/oxid-cookie-compliance/pull/15)
- removed concat asset include [`#14`](https://github.com/aggrosoft/oxid-cookie-compliance/pull/14)
- Bump ini from 1.3.5 to 1.3.8 [`#13`](https://github.com/aggrosoft/oxid-cookie-compliance/pull/13)
- Bump node-fetch from 2.6.0 to 2.6.1 [`#9`](https://github.com/aggrosoft/oxid-cookie-compliance/pull/9)
- missing lang entry [`#10`](https://github.com/aggrosoft/oxid-cookie-compliance/pull/10)
- Update README.md [`#11`](https://github.com/aggrosoft/oxid-cookie-compliance/pull/11)
- Create agcookiecompliance_lang.php [`3113370`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/3113370c9393be7717958b57f493335f217b7348)
- Create agcookiecompliance_lang.php [`55d24b5`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/55d24b5c32d6ca0166c569ef03f473046f122fb6)

#### [1.1.7](https://github.com/aggrosoft/oxid-cookie-compliance/compare/1.1.6...1.1.7)

> 2 October 2020

- Release 1.1.7 [`43145a8`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/43145a8c6c8afd91560c9aa38eee39f9f5d6e92e)
- cookieinfos table display bug [`0092e8f`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/0092e8f21f76354d2e43e532ea1f1ce7afa070fc)

#### [1.1.6](https://github.com/aggrosoft/oxid-cookie-compliance/compare/1.1.5...1.1.6)

> 29 September 2020

- Update CHANGELOG.md [`bf15d0c`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/bf15d0cb6aa15466e874ef2679612c9f8613bc6e)
- fix widget for ee [`24b4ffd`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/24b4ffdd4a16c4035d922a1b9604796618952b84)
- Update metadata.php [`acad1f0`](https://github.com/aggrosoft/oxid-cookie-compliance/commit/acad1f037850eadef0ff391659cf873d33f59003)

#### [1.1.5](https://github.com/aggrosoft/oxid-cookie-compliance/compare/1.1.4...1.1.5)

> 1 September 2020

- Netfeld [`#6`](https://github.com/aggrosoft/oxid-cookie-compliance/pull/6)

copy_this/modules/agcookiecompliance/metadata.php
<?php

$sMetadataVersion = '1.1';

$icon = '<svg style="width:11px;height:11px" viewBox="0 0 24 24">
    <path fill="#804700" d="M12,3A9,9 0 0,0 3,12A9,9 0 0,0 12,21A9,9 0 0,0 21,12C21,11.5 20.96,11 20.87,10.5C20.6,10 20,10 20,10H18V9C18,8 17,8 17,8H15V7C15,6 14,6 14,6H13V4C13,3 12,3 12,3M9.5,6A1.5,1.5 0 0,1 11,7.5A1.5,1.5 0 0,1 9.5,9A1.5,1.5 0 0,1 8,7.5A1.5,1.5 0 0,1 9.5,6M6.5,10A1.5,1.5 0 0,1 8,11.5A1.5,1.5 0 0,1 6.5,13A1.5,1.5 0 0,1 5,11.5A1.5,1.5 0 0,1 6.5,10M11.5,11A1.5,1.5 0 0,1 13,12.5A1.5,1.5 0 0,1 11.5,14A1.5,1.5 0 0,1 10,12.5A1.5,1.5 0 0,1 11.5,11M16.5,13A1.5,1.5 0 0,1 18,14.5A1.5,1.5 0 0,1 16.5,16H16.5A1.5,1.5 0 0,1 15,14.5H15A1.5,1.5 0 0,1 16.5,13M11,16A1.5,1.5 0 0,1 12.5,17.5A1.5,1.5 0 0,1 11,19A1.5,1.5 0 0,1 9.5,17.5A1.5,1.5 0 0,1 11,16Z" />
</svg>';

$aModule = array(
    'id'           => 'agcookiecompliance',
    'title'        => $icon . ' Cookie Compliance',
    'description'  => 'GDPR Cookie Compliance',
    'thumbnail'    => '',
    'version'      => '1.2.4',
    'author'       => 'Aggrosoft GmbH',
    'extend'      => array(
        'oxviewconfig' => 'agcookiecompliance/extensions/core/agcookiecompliance_oxviewconfig'
    ),
    'files'       => array(
        'agcookiecompliance_installer' => 'agcookiecompliance/application/core/agcookiecompliance_installer.php',
        'cookietrainer' => 'agcookiecompliance/application/controllers/cookietrainer.php',
        'compliancecookie' => 'agcookiecompliance/application/models/compliancecookie.php',
        'compliancecookielist' => 'agcookiecompliance/application/models/compliancecookielist.php',
        'cookieparser' => 'agcookiecompliance/application/core/cookieparser.php',
        'cookiehelper' => 'agcookiecompliance/application/core/cookiehelper.php',
        'admin_compliancecookies' => 'agcookiecompliance/application/controllers/admin/admin_compliancecookies.php',
        'compliancecookies_list' => 'agcookiecompliance/application/controllers/admin/compliancecookies_list.php',
        'compliancecookies_main' => 'agcookiecompliance/application/controllers/admin/compliancecookies_main.php',
    ),
    'templates'   => array(
        'widget/cookiecompliance.tpl' => 'agcookiecompliance/application/views/tpl/widget/cookiecompliance.tpl',
        'widget/cookiecompliancedialog.tpl' => 'agcookiecompliance/application/views/tpl/widget/cookiecompliancedialog.tpl',
        'widget/cookieinfos.tpl' => 'agcookiecompliance/application/views/tpl/widget/cookieinfos.tpl',
        'compliancecookies.tpl' => 'agcookiecompliance/application/views/admin/tpl/compliancecookies.tpl',
        'compliancecookies_list.tpl' => 'agcookiecompliance/application/views/admin/tpl/compliancecookies_list.tpl',
        'compliancecookies_main.tpl' => 'agcookiecompliance/application/views/admin/tpl/compliancecookies_main.tpl',
    ),
    'events'       => array(
        'onActivate'   => 'agcookiecompliance_installer::onActivate',
    ),
    'settings' => array(
        array('group' => 'agcookiecompliance_layout', 'name' => 'sBannerPosition', 'type' => 'select', 'value' => 'bottom', 'constraints' => 'bottom|top|bottom-left|bottom-right'),
        array('group' => 'agcookiecompliance_layout', 'name' => 'blStaticBanner', 'type' => 'bool', 'value' => false),
        array('group' => 'agcookiecompliance_layout', 'name' => 'sBannerTheme', 'type' => 'select', 'value' => 'block', 'constraints' => 'block|edgeless|classic|custom'),
        array('group' => 'agcookiecompliance_colors', 'name' => 'sBannerBackgroundColor', 'type' => 'str', 'value' => '#eaf7f7'),
        array('group' => 'agcookiecompliance_colors', 'name' => 'sBannerTextColor', 'type' => 'str', 'value' => '#5c7291'),
        array('group' => 'agcookiecompliance_colors', 'name' => 'sButtonBackgroundColor', 'type' => 'str', 'value' => '#56cbdb'),
        array('group' => 'agcookiecompliance_colors', 'name' => 'sButtonTextColor', 'type' => 'str', 'value' => '#ffffff'),
        array('group' => 'agcookiecompliance_colors', 'name' => 'sHighlightBackgroundColor', 'type' => 'str', 'value' => '#d65959'),
        array('group' => 'agcookiecompliance_colors', 'name' => 'sHighlightTextColor', 'type' => 'str', 'value' => '#ffffff'),
        array('group' => 'agcookiecompliance_colors', 'name' => 'sHighlightBorderColor', 'type' => 'str', 'value' => '#d65959'),
        array('group' => 'agcookiecompliance_settings', 'name' => 'sConsentType', 'type' => 'select', 'value' => 'categories', 'constraints' => 'categories|opt-in|opt-out|info'),
        array('group' => 'agcookiecompliance_settings', 'name' => 'sInfoLink', 'type' => 'str', 'value' => ''),
        array('group' => 'agcookiecompliance_settings', 'name' => 'blTrainingMode', 'type' => 'bool', 'value' => false),
        array('group' => 'agcookiecompliance_settings', 'name' => 'blJqueryMode', 'type' => 'bool', 'value' => true),
    ),
    'blocks' => array(
        array('template' => 'layout/base.tpl','block'=>'base_js','file'=>'/views/blocks/base_js.tpl'),
        array('template' => 'layout/base.tpl','block'=>'base_style','file'=>'/views/blocks/base_style.tpl'),
    )
);
